gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
You're right - I was thinking mainly in terms of VehiclePositions (and hence only considered the **next** station). However, TripUpdates can specify **many** stations, so we need info about the carriage directions for each of them.

I still think it's important to know what both the arriving and departing sequence is at a particular station.

Revising:
- `CarriageDescriptor` still sits in `VehicleDescriptor`, but only contains info about the carriage facilities (no sequencing info)
- TUs: `StopTimeUpdates` contain the carriage sequences
- VPs: the carriage sequence is at the top level, and is for the stop specified in `stop_id`

More concretely:
```
message VehicleDescriptor {
  ....
  repeated CarriageDescriptor carriage_descriptor;
  message CarriageDescriptor {
    optional string id = 1;
    ... wifi, occupancy, bicycles etc
  }
}

message CarriageSequence {
  optional string carriage_descriptor_id = 1;
  optional int sequence = 2;
}

message TripUpdate {
  ...
  message StopTimeUpdate {
    ...
    optional string stop_id = 4;
    ...
    repeated CarriageSequence arrival_carriage_sequence = X;
    repeated CarriageSequence departure_carriage_sequence = X+1;
  }
}

message VehiclePosition {
  ...
  optional string stop_id = 7;
  ...
  repeated CarriageSequence arrival_carriage_sequence = X;
  repeated CarriageSequence departure_carriage_sequence = X+1;
}
```